lib/appraisal/bundler_dsl.rb

```diff
@@ -0,0 +1,121 @@
+require 'appraisal/dependency_list'
+
+module Appraisal
+  class BundlerDSL
+    attr_reader :dependencies
+
+    def initialize
+      @sources = []
+      @ruby_version = nil
+      @dependencies = DependencyList.new
+      @gemspec = nil
+      @groups = {}
+      @platforms = {}
+      @git_sources = {}
+      @path_sources = {}
+    end
+
+    def run(&block)
+      instance_exec(&block)
+    end
+
+    def gem(name, *requirements)
+      @dependencies.add(name, requirements)
+    end
+
+    def group(*names, &block)
+      @groups[names] ||= Group.new(names)
+      @groups[names].run(&block)
+    end
+
+    # :nodoc:
+    def groups(*names, &block)
+      $stderr.puts <<-WARNING.gsub(/\n\s+/, " ").strip
+        Warning: `#groups` is deprecated and will be removed in 2.0.0.
+        Please use `#group` instead.
+      WARNING
+
+      group(*names, &block)
+    end
+
+    def platforms(*names, &block)
+      @platforms[names] ||= Platform.new(names)
+      @platforms[names].run(&block)
+    end
+
+    alias_method :platform, :platforms
+
+    def source(source)
+      @sources << source
+    end
+
+    def ruby(ruby_version)
+      @ruby_version = ruby_version
+    end
+
+    def git(source, options = {}, &block)
+      @git_sources[source] ||= GitSource.new(source, options)
+      @git_sources[source].run(&block)
+    end
+
+    def path(source, options = {}, &block)
+      @path_sources[source] ||= PathSource.new(source, options)
+      @path_sources[source].run(&block)
+    end
+
+    def to_s
+      [source_entry,
+        ruby_version_entry,
+        git_sources_entry,
+        path_sources_entry,
+        dependencies_entry,
+        groups_entry,
+        platforms_entry,
+        gemspec_entry].reject{ |s| s.nil? || s.empty? }.join("\n\n").strip
```

Comment:
Align the elements of an array literal if they span more than one line.<br>Space missing to the left of {.